CompoundV3-Bots/Reserves-Monitor/src/agent.spec.ts

```diff
@@ -0,0 +1,337 @@
+    alertFrequency: ALERT_FREQ,
+  };
+  const fetcher = new Fetcher(mockProvider as any, mockNetworkManager as any);
+
+  beforeAll(async () => {
+    await fetcher.setContracts();
+  });
+
+  beforeEach(async () => {
+    jest.resetModules();
+    const BOT = require("./agent"); // Done to re-initialize ALERTS struct.
+    handleBlock = BOT.provideHandleBlock(fetcher);
+  });
+
+  it("returns empty findings if reserves are less than target reserves", async () => {
+    const blockEvent: BlockEvent = new TestBlockEvent()
+      .setNumber(10)
+      .setTimestamp(INIT_TIMESTAMP);
+    for (let addr of TEST_ADDRESSES) {
+      createTargetReservesCall(addr, BigNumber.from(100));
+      createGetReservesCall(addr, BigNumber.from(80));
+    }
+
+    const findings = await handleBlock(blockEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns empty findings if reserves are negative", async () => {
+    const blockEvent: BlockEvent = new TestBlockEvent()
+      .setNumber(10)
+      .setTimestamp(INIT_TIMESTAMP);
+
+    for (let addr of TEST_ADDRESSES) {
+      createTargetReservesCall(addr, BigNumber.from(100));
+      createGetReservesCall(addr, BigNumber.from(-200));
+    }
+    const findings = await handleBlock(blockEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns a finding if Reserves == targetReserves in one comet contract", async () => {
+    let reservesValue: BigNumber = BigNumber.from(200);
+    const blockEvent: BlockEvent = new TestBlockEvent()
+      .setNumber(10)
+      .setTimestamp(INIT_TIMESTAMP);
+
+    for (let i = 1; i < TEST_ADDRESSES.length; i++) {
+      createTargetReservesCall(TEST_ADDRESSES[i], BigNumber.from(100));
+      createGetReservesCall(TEST_ADDRESSES[i], BigNumber.from(50));
+    }
+    createTargetReservesCall(testConfig.cometAddresses[0], reservesValue);
+    createGetReservesCall(testConfig.cometAddresses[0], reservesValue);
+
+    const findings = await handleBlock(blockEvent);
+    expect(JSON.stringify(findings)).toStrictEqual(
+      JSON.stringify([
+        Finding.from({
+          name: `Comet reserves reached targetReserves`,
+          description: `Reserves on comet contract are >= target reserves`,
+          alertId: "COMP2-1",
+          protocol: "Compound",
+          severity: FindingSeverity.Medium,
+          type: FindingType.Info,
+          metadata: {
+            comet: testConfig.cometAddresses[0],
+            reserves: reservesValue.toString(),
+            targetReserves: reservesValue.toString(),
+          },
+        }),
+      ])
+    );
+  });
+
+  it("returns a finding if Reserves > targetReserves in one comet contract", async () => {
+    let reservesValue: BigNumber = BigNumber.from(500);
+    let targetReservesValue: BigNumber = BigNumber.from(200);
+
+    const blockEvent: BlockEvent = new TestBlockEvent()
+      .setNumber(10)
+      .setTimestamp(INIT_TIMESTAMP);
+
+    for (let i = 1; i < TEST_ADDRESSES.length; i++) {
```

Comment:
Same suggestion as above, just to keep the `for...of` syntax and do it in the array order instead of setting stuff for the first element last.

Reply:
Applies for all similar cases